platform/binance/api.go

```diff
@@ -25,37 +25,53 @@ func Setup(router gin.IRouter) {
 	router.GET("/:address", getTransactions)
 }
 
+func extractBase(srcTx *source.Tx) (base models.Tx, ok bool) {
+	var err error
+	date, err := time.Parse("2006-01-02T15:04:05.999Z", srcTx.Timestamp)
+	unix := date.Unix()
+	if err != nil {
+		return base, false
+	}
+
+	base = models.Tx{
+		Id:    srcTx.Hash,
+		Coin:  coin.BNB,
+		Date:  unix,
+		From:  srcTx.FromAddr,
+		To:    srcTx.ToAddr,
+		Fee:   srcTx.Fee,
+		Block: srcTx.BlockHeight,
+	}
+	return base, true
+}
+
 func getTransactions(c *gin.Context) {
 	s, err := client.GetTxsOfAddress(c.Param("address"))
 	if apiError(c, err) {
 		return
 	}
 
 	var txs []models.Tx
-	for _, tx := range s.Txs {
-		if tx.Asset != "BNB" {
+	for _, srcTx := range s.Txs {
+		if srcTx.Asset != "BNB" {
 			continue
 		}
 
-		var err error
-		date, err := time.Parse("2006-01-02T15:04:05.999Z", tx.Timestamp)
-		unix := date.Unix()
-		if err != nil {
-			unix = 0
+		tx, ok := extractBase(&srcTx)
+		if !ok {
+			continue
 		}
-
-		txs = append(txs, models.Tx{
-			Id:    tx.Hash,
-			Coin:  coin.BNB,
-			Date:  unix,
-			From:  tx.FromAddr,
-			To:    tx.ToAddr,
-			Fee:   tx.Fee,
-			Block: tx.BlockHeight,
-			Meta:  models.Transfer{
-				Value: tx.Value,
-			},
-		})
+		if srcTx.Asset == "BNB" {
+			tx.Meta = models.Transfer{
+				Value: srcTx.Value,
+			}
+		} else {
+			tx.Meta = models.TokenTransfer{
```

Comment:
@terorie can you use new type `native_token_transfer` ? but same structure as `token_transfer` this would mean it's not token transfer by the smart contract. iOS version uses this model already

Reply:
Done